merkle-tree/concurrent/src/lib.rs

```diff
@@ -623,23 +616,15 @@ where
                 }
 
                 if i < self.height - 1 {
-                    self.changelog[changelog_index].path[i + 1] = current_node;
-
-                    for leaf_j in 0..leaf_i {
-                        let changelog_index =
-                            (first_changelog_index + leaf_j) % self.changelog.capacity();
-                        if self.changelog[changelog_index].path[i + 1] == [0u8; 32] {
-                            self.changelog[changelog_index].path[i + 1] = current_node;
-                        }
-                    }
+                    self.changelog[changelog_index].path[i + 1] = Some(current_node);
                 }
 
                 current_index /= 2;
             }
 
-            self.changelog[changelog_index].root = current_node;
-
-            self.roots.push(current_node);
+            if leaf_i == leaves.len() - 1 {
+                self.roots.push(current_node);
+            }
```

Comment:
```suggestion
            } else {
             self.push([0u8;32]);
            }
```
We need to push the same number of roots as we append leaves.
Photon only returns the sequence number and we use it in the js client to derive the root index.
This is likely the reason why cli tests fail.

Reply:
If we need to push the same number of roots, then we either need to:

- push the same root N times (feels really stupid...)
- compute the whole Merkle paths (we lose performance)

I guess you had the 1st solution in mind?

Reply:
Done